### PR TITLE
Finishing upgrade to Kirkstone

### PR DESCRIPTION
**Pull request by nihemstr**
- Updated keywords
- Removed old design patterns 
- Updated the build configurations to target only kirkstone

### CODE REVIEW

**Review by jw-msft — APPROVED** [review]
Approved with comment

File: azurepipelines/adu-yocto-build.yml [hunk 1]
```diff
@@ -15,84 +15,68 @@
...
   default: develop
   values:
-    - main
     - develop
+
 - name: adu_git_commit
   displayName: ADU Git commit 
   type: string
-  default: e099aaca0ad2b8849de24a13b1aa8e3ddf009251
+  default: 60bb98ae3631419b393c528f7dc3cf0797b231e6
```
Comment (jw-msft) [inline azurepipelines/adu-yocto-build.yml:41]: should it be latest commit which is [10791ac6bd98eaf4e87ea54350d3e9c8a2cec1da ](https://github.com/Azure/iot-hub-device-update/commit/10791ac6bd98eaf4e87ea54350d3e9c8a2cec1da) now?

File: azurepipelines/build-template.yml [hunk 2]
```diff
@@ -2,67 +2,56 @@ jobs:
   - job: build_yocto
     displayName: Build Yocto images
     pool:
-      name: 1es_hosted_yocto_build
+      name: 1es_hosted_pool_yocto
```
Comment (Jerry-GeGarcia) [inline azurepipelines/build-template.yml:5]: we do run in a 1es hosted pool, and don't time out. Cool

File: azurepipelines/build-template.yml [hunk 3]
```diff
@@ -2,67 +2,56 @@ jobs:
...
         
       - task: Bash@3
         displayName: "Run Yocto build"
         inputs:
           targetType: "filePath"
-          filePath: "$(Pipeline.Workspace)/adu-yocto-build/scripts/build.sh"
-          arguments: '-c -t $(build_type) -v $(Build.BuildNumber) --adu-src-uri $(adu_src_uri) --adu-git-branch $(adu_git_branch) --adu-git-commit $(adu_git_commit) --do-src-uri $(do_src_uri) --do-git-branch $(do_git_branch)  --do-git-commit $(do_git_commit) --adu-delta-src-uri $(adu_delta_src_uri) --adu-delta-git-branch $(adu_delta_git_branch) --adu-delta-git-commit $(adu_delta_git_commit) -o $(Build.BinariesDirectory) $(build_args)'
-          workingDirectory: $(Pipeline.Workspace)
+          filePath: "./scripts/build.sh"
+          arguments: '-c -t MinSizeRel -v $(Build.BuildNumber) --adu-src-uri $(adu_src_uri) --adu-git-branch $(adu_git_branch) --adu-git-commit $(adu_git_commit) --do-src-uri $(do_src_uri) --do-git-branch $(do_git_branch)  --do-git-commit $(do_git_commit) --adu-delta-src-uri $(adu_delta_src_uri) --adu-delta-git-branch $(adu_delta_git_branch) --adu-delta-git-commit $(adu_delta_git_commit) -o $(Build.BinariesDirectory) $(build_args)'
```
Comment (Jerry-GeGarcia) [inline azurepipelines/build-template.yml:53]: cool, we were not optimizing our build before :(

File: yocto/config-templates/raspberrypi3/local.conf.sample [hunk 4]
```diff
@@ -151,12 +151,7 @@ EXTRA_IMAGE_FEATURES ?= "debug-tweaks tools-debug tools-sdk package-management"
 # are:
 #   - 'buildstats' collect build statistics
 
-# The ``image-prelink`` class enables the use of the ``prelink`` utility
```
Comment (Jerry-GeGarcia) [inline yocto/config-templates/raspberrypi3/local.conf.sample:154]: we didn't need this anymore?

File: azurepipelines/adu-yocto-build.yml (hunk 1 above)
Comment (Jerry-GeGarcia) [reply azurepipelines/adu-yocto-build.yml:41]: pls double check

**Review by Jerry-GeGarcia — APPROVED** [review]

File: azurepipelines/adu-yocto-build.yml (hunk 1 above)
Comment (nihemstr) [reply azurepipelines/adu-yocto-build.yml:41]: These will have to be updated in a release.

**Pull request merged**